### DISCUSSION TITLE
Double click

### DISCUSSION THREAD

**Discussion by Schwenkner** [arendst/Tasmota#10491] [Q&A] [answered]
I have a question upon the double click behaviour.

When I flashed my Gosund SP112, I can turn on and off the USB ports with pressing the button twice. If pressed only once, the power is switched.

Now I have a Gosund P1 which also has Usb ports. There I cannot switch them with double click. Instead I could apply a rule, but as described, then also two single clicks will be fired.

My question is, how the double click in the SP112 is realized, it seems to be not a rule?
And can this be applied to the P1?

**Comment by barbudor** [comment, marked as answer]
Hi @Schwenkner 
The "Button1" has a special behavior when multiple relays are configured
Single press toogles Power1
Double press toggles Relay2
and so on
You should have the same behavior with the P1 if you don't apply the rules provided in the template

The Rules provided in the template, as per the text, will modify the default behavior to have main button assign to USB, and smaller buttons assigned to each socket.

So in short : if you prefer multi-press to control your strip, do not use any of the rules provided in the template.

**Comment by Schwenkner** [comment]
Thank you for this explanation, nowhere else read about this behaviour.

Did I understood correct, pressing3 times will toggle 3rd relay and 4 times then USB?
After clean flashing the P1 didn't try this, got only first relay toggled. This explains all.

But then there is another question. As I understand rule 2, this is necessary to toggle the relays with the small buttons, aren't they?
I remember after the clean flash, I couldn't toggle the relays with the small buttons unless I aplied rule 2.

Multi-press behaviour would be good, but I also will have the small buttons working

And as I remeber well, the original Gosund firmware turned off all releays when at least one was on when pressing the main button and turned all relays on, when none has been on before.

It is ok for me, when Tasmota doesn't behave the same way as the original, I only will try to understand it.

**Comment by barbudor** [comment]
it is documented on the "Buttons and Switches" page : https://tasmota.github.io/docs/Buttons-and-Switches/#multi-press-functions
See the note, and the warning above

Yes, Rule2 is used to toggle relays individually using the small buttons.
The person who wrote ths template decided to assign the big button to the USB and use small buttons for each matching relays.

You can have both multipress (if not using Rule1) and also keep Rule2 for small buttons.

The original Gosund behavior is possible but slightly more complex to setup with rules.
